microprofile/lra/jax-rs/src/test/java/io/helidon/microprofile/lra/ParticipantTest.java

```diff
@@ -17,45 +17,180 @@
+                    req.content()
+                            .as(String.class)
+                            .flatMap(s -> Multi.create(Arrays.stream(s.split(","))))
+                            .map(s -> s.split(";")[0])
+                            .map(s -> s
+                                    .replaceAll("^<", "")
+                                    .replaceAll(">$", "")
+                            )
+                            .map(URI::create)
+                            .map(URI::getPath)
+                            .onComplete(res::send)
+                            .onComplete(() -> completed.complete(null))
+                            .forEach(paths::add)
+                            .exceptionally(res::send);
+                });
     }
 
-    void logFine(String msg, Supplier<List<?>> params) {
-        if (LOGGER.isLoggable(Level.FINE)) {
-            LOGGER.log(Level.FINE, msg, params.get().toArray());
-        }
+    private void ready(
+            @Observes
+            @Priority(PLATFORM_AFTER + 101)
+            @Initialized(ApplicationScoped.class) Object event,
+            BeanManager bm) {
+        port = bm.getExtension(ServerCdiExtension.class).port();
+        // Provide LRA client with coordinator loadbalancer url
+        coordinatorLocatorService.overrideCoordinatorUriSupplier(() ->
+                URI.create("http://localhost:" + port + "/lra-coordinator"));
+    }
+
+    @Test
+    void configurableContext(WebTarget target) throws Exception {
+        Response response = target.path("participant-test")
+                .path("start")
+                .request()
+                .header(Work.HEADER_KEY, Work.NOOP)
+                .async()
+                .put(Entity.text(""))
+                .get(TIMEOUT_SEC, TimeUnit.SECONDS);
+
+        assertThat(response.getStatus(), AnyOf.anyOf(is(200), is(204)));
+
+        completed.get(TIMEOUT_SEC, TimeUnit.SECONDS);
+
+        assertThat(paths.size(), is(2));
+        paths.forEach(path -> assertThat(path, startsWith("/" + CUSTOM_CONTEXT)));
     }
 
     @Test
     void methodScan() throws NoSuchMethodException {
-        ParticipantImpl p = new ParticipantImpl(UriBuilder.fromPath("http://localhost:8888").build(), DontEnd.class);
+        ParticipantImpl p = new ParticipantImpl(
+                URI.create("http://localhost:8888"),
+                NonJaxRsResource.CONTEXT_PATH_DEFAULT,
+                DontEnd.class);
         assertTrue(p.isLraMethod(DontEnd.class.getMethod("startDontEndLRA", URI.class)));
         assertTrue(p.isLraMethod(DontEnd.class.getMethod("endLRA", URI.class)));
     }
+
+    @ApplicationScoped
+    @Path("/participant-test")
+    public static class TestResource {
+
+        @PUT
+        @LRA(value = LRA.Type.REQUIRES_NEW)
+        @Path("/start")
+        public void start(
+                @HeaderParam(LRA_HTTP_CONTEXT_HEADER) URI lraId,
+                @HeaderParam(Work.HEADER_KEY) Work work
```

Comment:
please put `) {` at the end of previous line

Reply:
fixed